docs/users/topics/phpstorm.md

```diff
@@ -0,0 +1,50 @@
+2. Set up your project to do normal Xdebug, as described in the [Step Debugging section](../step_debugging.md). This will result in a PhpStorm "Server" with the proper name, normally the same as the FQDN of the project. In this example, "d9.ddev.site". (All you have to do here is click the little telephone to "Start listening for PHP Debug Connections", then `ddev xdebug on`, then visit a web page and choose the correct mapping from host to server. )
+3. Under File→Settings→PHP (Windows) or Preferences→PHP (macOS), click the "..." to the right of "CLI Interpreter"
+    1. Use the "+" to select "From Docker, Vagrant, VM..."
+    2. Choose "Docker Compose"
+    3. Create a server; the default name is "docker", but since the "server" for each project will be different, name it for the project, for example "DDEV d9". Choose "Docker for Windows" or "Docker for Mac"
+    4. In the "Path mappings" of the "Server" map the local paths (which on WSL2 means /home/...) to the in-container paths. So "Virtual Machine Path" would be "/var/www/html" and "Local path" would be something like /Users/rfay/workspace/d9 (on macOS) or \\wsl$\Ubuntu\home\rfay\workspace\d9 on Windows using WSL2
+    5. Now back in the "Configure Remote PHP Interpreter" for "Configuration files" use .ddev/.ddev-docker-compose-full.yaml
+    6. Service: web
+    7. Add an environment variable `COMPOSE_PROJECT_NAME=ddev-<projectname>`. In my case, it's `ddev-d9`
+    8. In the CLI interpreter "Lifecycle" select "Connect to existing container"
+    9. In the PHP Interpreter path, you can just put "php" if you're using the default PHP version (currently 7.4). Due to a [PhpStorm bug](https://youtrack.jetbrains.com/issue/WI-62463) you'll want to put the full name of the binary, like "php8.0" if you're not using the default version.
+    10. Here's an example filled out ![example configuration](images/cli_interpreter.png)
+4. In the main PHP setup dialog, add an entry to the path mappings, as it doesn't correctly derive the full path mapping. Add an entry that maps your project location to /var/www/html. So in this example, the Local Path is /Users/rfay/workspace/d9 and the Remote Path is /var/www/html. ![example mapping](images/mapping.png)
+5. Configure composer under PHP→Composer.
+    - Use "remote interpreter"
+    - CLI Interpreter will be "web"
+6. Under "Test Frameworks" click the "+" to add phpunit
+    - PHPUnit by remote interpreter
+    - Interpreter "web"
+    - Choose "Path to phpunit.phar" and use /var/www/html/vendor/bin/phpunit (or wherever your phpunit is inside the container) (You need phpunit properly composer-installed for your CMS. For example, for Drupal 9, `ddev composer require --dev --with-all-dependencies drupal/core-dev:^9`  and `ddev composer require --dev phpspec/prophecy-phpunit:^2`
```

Comment:
Is this better than using "Use composer autoloader" and putting in vendor/autoload.php?

Reply:
When I tested this time, I couldn't get "Use composer autoloader" to work, but I have seen it work before.

Reply:
But there's nothing wrong with the other way, of course... if it works.

Reply:
Using the autoloader did work well for me. 
Maybe adding some sort of FAQ section instead with "PHPStorm can not auto detect my phpunit version" and as fix these steps?

Reply:
Suggestions welcome. (Use the `Insert a suggestion` on comments.). Not sure we have to solve every problem or provide every option.

Reply:
I think the current steps are fine.

Reply:
I think it's obvious this will evolve as time goes on, it doesn't have to be perfect right now, and people can contribute improvements as we get experience. And as PhpStorm fixes things up.